src/sentry/incidents/endpoints/organization_alert_rule_details.py

```diff
@@ -114,7 +118,16 @@ def remove_alert_rule(request: Request, organization, alert_rule):
         # NOTE: we want to run the dual delete regardless of whether the user is flagged into dual writes:
         # the user could be removed from the dual write flag for whatever reason, and we need to make sure
         # that the extra table data is deleted. If the rows don't exist, we'll exit early.
-        dual_delete_migrated_alert_rule(alert_rule=alert_rule, user=request.user)
+        try:
+            dual_delete_migrated_alert_rule(alert_rule=alert_rule, user=request.user)
+        except Exception as e:
+            logger.exception(
+                "Error when dual deleting alert rule",
+                extra={"details": str(e)},
+            )
+            return Response(
+                "Error when dual deleting alert rule", status=status.HTTP_400_BAD_REQUEST
```

Comment:
We may want to look at the kind of exception to decide the status code here -- [400 errors](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400) are mean that they're caused by bad user input.

Since there are a lot of possible errors that can arise when we are deleting these rules, we should check the exception to see if it was caused by user input (4xx errors) or something on our end (5xx errors).

side note for the best of the http status codes, [418](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/418)